pkg/quota/evaluator/core/pods.go

```diff
@@ -244,14 +245,21 @@ func podComputeUsageHelper(requests api.ResourceList, limits api.ResourceList) a
 		result[api.ResourceLimitsEphemeralStorage] = limit
 	}
 	for resource, request := range requests {
+		// for resources with certain prefix, e.g. hugepages
 		if quota.ContainsPrefix(requestedResourcePrefixes, resource) {
 			result[resource] = request
-			result[maskResourceWithPrefix(resource, requestsPrefix)] = request
+			result[maskResourceWithPrefix(resource, api.DefaultResourceRequestsPrefix)] = request
+		}
+		// for extended resources
+		if helper.IsExtendedResourceName(resource) {
+			// only quota objects in format of "requests.resourceName" is allowed for extended resource.
+			result[maskResourceWithPrefix(resource, api.DefaultResourceRequestsPrefix)] = request
 		}
 	}
 	for resource, limit := range limits {
+		// for resources with certain prefix, e.g. hugepages
 		if quota.ContainsPrefix(requestedResourcePrefixes, resource) {
-			result[maskResourceWithPrefix(resource, limitsPrefix)] = limit
+			result[maskResourceWithPrefix(resource, api.DefaultResourceLimitsPrefix)] = limit
```

Comment:
i think this code block can be removed.

if we only allow quota on request.hugepages, we don't need to report limit usage... either way, its unused and should get stripped since it wont match any valid input quota from a user.

Reply:
I meant to build a common mechanism for potential resources with specific prefix.
Would remove this along with the constant statement since we could not see such kind of resources that allow overcommit so far.